capellambse/model/common/accessors.py

```diff
@@ -1201,16 +1272,23 @@ def __setitem__(
         self, index: int | slice, value: T | cabc.Iterable[T]
     ) -> None:
         assert self._parent is not None
-        del self[index]
-        if isinstance(index, slice):
-            assert isinstance(value, cabc.Iterable)
-            for i, elm in enumerate(value, start=index.start or 0):
-                self.insert(i, elm)
-        else:
-            assert not isinstance(value, cabc.Iterable)
-            self.insert(index, value)
+        accessor = type(self)._accessor
+        assert isinstance(accessor, WritableAccessor)
+
+        new_objs = list(self)
+        new_objs[index] = value  # type: ignore[index]
+
+        if self.fixed_length and len(new_objs) != self.fixed_length:
+            raise TypeError(
+                f"Cannot set: List must stay at length {self.fixed_length}"
+            )
+
+        accessor.__set__(self._parent, new_objs)
 
     def __delitem__(self, index: int | slice) -> None:
+        if self.fixed_length and len(self) <= self.fixed_length:
+            raise TypeError("Cannot delete from a fixed-length list")
```

Comment:
`TypeError` seems weird to me. The error condition is that this `ElementList` has a `fixed_length` and removing from it would cause a length violation: There's one less element in it opposing the constraint.

I'd expect either a custom exception or a `ValueError` here too with a different message. Deletion can be done if somehow there are more elements written in the XML via Capella and `len(self) >= self.fixed_length + 1`. Else why is there functional code afterwards?

Reply:
> Deletion can be done if somehow there are more elements written in the XML via Capella and `len(self) >= self.fixed_length + 1`. Else why is there functional code afterwards?

Items can also be deleted if the list isn't `fixed_length` to begin with, which would be the typical case to hit that code.